metpy/calc/indices.py

```diff
@@ -51,3 +51,48 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, hgt=None, depth=None, bottom=None):
+    r"""Calculate pressure-weighted mean wind through a layer.
+
+    Layer top and bottom specified in height or pressure.
+
+    Parameters
+    ----------
+    u : `pint.Quantity`
+        U-component of wind.
+    v : `pint.Quantity`
+        V-component of wind.
+    p : `pint.Quantity`
+        Atmospheric pressure profile
+    hgt : `pint.Quantity`
```

Comment:
`hgt` -> `heights`, both to avoid unneeded shortening and for consistency with e.g. `get_layer`